### PR TITLE
Add sort by name/id functionality

### LINKED ISSUE #68: Add feature to sort lists of students by name and student ID.
As a teacher I would like to be able to sort my students by student ID and name.

### PR DESCRIPTION
**Pull request by IamZhenHong**
fixes #68

### CODE REVIEW

File: src/main/java/seedu/address/model/person/StudentId.java [hunk 1]
```diff
@@ -27,6 +27,11 @@ public static boolean isValidStudentId(String test) {
         return test.matches(VALIDATION_REGEX);
     }
 
+    @Override
+    public int compareTo(StudentId other) {
+        return Integer.valueOf(value).compareTo(Integer.valueOf(other.value));
```
Comment (ongzhili) [inline src/main/java/seedu/address/model/person/StudentId.java:32]: Can be tested

File: src/main/java/seedu/address/logic/commands/SortCommand.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                    if (name1.charAt(i) != name2.charAt(i)) {
+                        return name1.charAt(i) - name2.charAt(i);
+                    }
+                }
+                // If all characters are the same up to the length of the shorter string,
+                // the shorter string comes first.
+                return name1.length() - name2.length();
+            }
+            return result;
+        }
```
Comment (ongzhili) [inline src/main/java/seedu/address/logic/commands/SortCommand.java:56]: Comparison metric has many levels of testing possible - Can extensively test i think

File: src/main/java/seedu/address/logic/commands/SortCommand.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        switch (sortType) {
+        case "id":
+            model.sortFilteredPersonList(PERSON_ID_COMPARATOR);
+            return new CommandResult(String.format(Messages.MESSAGE_LIST_SORTED_SUCCESSFULLY, "id"));
+        case "name":
+            model.sortFilteredPersonList(PERSON_NAME_COMPARATOR);
+            return new CommandResult(String.format(Messages.MESSAGE_LIST_SORTED_SUCCESSFULLY, "name"));
+        default:
+            throw new CommandException(Messages.MESSAGE_INVALID_COMMAND_FORMAT);
+        }
```
Comment (ongzhili) [inline src/main/java/seedu/address/logic/commands/SortCommand.java:80]: Same here. Adding a testcase with some nonsense sortType to catch the invalid command format recommended

**Review by ongzhili — CHANGES_REQUESTED** [review]
Can be tested more. Otherwise code looks fine I think.

File: src/main/java/seedu/address/logic/commands/SortCommand.java (hunk 3 above)
Comment (IamZhenHong) [reply src/main/java/seedu/address/logic/commands/SortCommand.java:80]: Good suggestion.Added

File: src/main/java/seedu/address/logic/commands/SortCommand.java (hunk 2 above)
Comment (IamZhenHong) [reply src/main/java/seedu/address/logic/commands/SortCommand.java:56]: Yes, didn't think of using this, good thinking.

**Review by ongzhili — APPROVED** [review]
LGTM

**Pull request merged**